src/main/java/io/github/kawamuray/wasmtime/TrapException.java

```diff
@@ -0,0 +1,12 @@
+package io.github.kawamuray.wasmtime;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+
+@AllArgsConstructor
+@Accessors(fluent = true)
+public class TrapException extends RuntimeException {
```

Comment:
I think it's a common and a good practice to make all exceptions a descendant of `WasmtimeException` so that users can selectively catch exception at arbitrary level of abstraction, such that:
* catch `Exception` or (`RuntimeException`) if they care only about "error or not" or
* catch `WasmtimeException` if they care any failure occurred at wasmtime runtime level or
* catch `TrapException` if they want to do something specific about trap